src/main/java/com/legendary/coffeeShop/dao/repositories/ComponentRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.legendary.coffeeShop.dao.repositories;
+
+import com.legendary.coffeeShop.dao.entities.Component;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Set;
+
+public interface ComponentRepository extends JpaRepository<Component, Long> {
+    Set<Component> findByProductTypesProductTypeContains(String type);
```

Comment:
This function should return all components that related to some product...am I right? 
if so... it should be like 
Set<Component> findAllByStatusAndProductTypesContains(ComponentStatus status, String type);

Reply:
why by status? if its to show only active components so thats a design desicion - from what I thought, we want to show active & out of stock components, only out of stock components should be greyed out or something (cause when you buy something online, there is OUT OF STOCK label. Cause lets say there is no tomatoes, it will be weird you won't see it on the site, user won't understand why

Reply:
You right but we don't want to return the component that discarded so change it to find by a list of statuses, BTW i just read that the best practice in spring repository is using List instead of Set and also to add distinct

Reply:
I think thats really weird to get list of statuses.. if you ensist - we can get by status and call twice (one for active and one for out of stock) and I think its also ok to get all and than the client will not show the discarded ones